### PR TITLE
[relay-runtime] add RelayNetworkLogger types

### PR DESCRIPTION
**Pull request by renanmav**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. ([RelayNetworkLoggerTransaction](https://github.com/facebook/relay/blob/master/packages/relay-runtime/network/RelayNetworkLoggerTransaction.js) and [createRelayNetworkLogger](https://github.com/facebook/relay/blob/master/packages/relay-runtime/network/createRelayNetworkLogger.js) of v6)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by renanmav** [comment]
Submited this PR to use RelayNetworkLogger. Currently, I'm doing this way:
```ts
import {
  Environment,
  Network,
  RecordSource,
  Store,
  // @ts-ignore
  RelayNetworkLoggerTransaction,
  // @ts-ignore
  createRelayNetworkLogger,
} from 'relay-runtime'
```
when working with `relay-runtime@6.0.0`.

**Review by sibelius — APPROVED** [review]

**Comment by renanmav** [comment]
Working locally.

```ts
import {
  Environment,
  Network,
  RecordSource,
  Store,
  RelayNetworkLoggerTransaction,
  createRelayNetworkLogger,
} from 'relay-runtime'

import cacheHandler from './cacheHandler'

const RelayNetworkLogger = createRelayNetworkLogger(RelayNetworkLoggerTransaction)

const network = Network.create(
  process.env.NODE_ENV === 'development' 
    ? RelayNetworkLogger.wrapFetch(cacheHandler) 
    : cacheHandler
)
```

File: types/relay-runtime/index.d.ts [hunk 1]
```diff
@@ -114,8 +115,30 @@ export interface MutationConfig<TOperation extends OperationType> {
 // ./network/RelayNetworkLoggerTransaction
 export interface RelayNetworkLog {
     label: string;
-    values: ReadonlyArray<unknown>;
+    values: ReadonlyArray<any>;
```
Comment (alloy) [inline types/relay-runtime/index.d.ts]: Why is this change necessary?

**Review by alloy — CHANGES_REQUESTED** [review]
Thanks for this!

3 things before we can merge this:
* The CI failure is legit.
* Can you add your snippet of how you use it as a test case please?
* Are any of these typings relay-runtime v6 _only_, or were they supported in v5 as well? If not, then let's not bump the version in the header.

**Comment by alloy** [comment]
Actually, never mind the question about bumping the version header, I'm already doing so here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38536

**Comment by alloy** [comment]
Btw, like I've done in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38536, would you mind moving your additions and existing related typings into a separate file?

File: types/relay-runtime/index.d.ts (hunk 1 above)
Comment (renanmav) [reply types/relay-runtime/index.d.ts]: Just following the typing in source code of relay

https://github.com/facebook/relay/blob/master/packages/relay-runtime/network/RelayNetworkLoggerTransaction.js#L46

**Comment by renanmav** [comment]
Sure, I'll move into a separate file.

The CI failure its because I was not able to clone the repo locally (problems with my internet). I'll try to fix it.

File: types/relay-runtime/index.d.ts (hunk 1 above)
Comment (alloy) [reply types/relay-runtime/index.d.ts]: Fair enough 👍

**Comment by renanmav** [comment]
How can I add tests to it?

Sorry, this is my first PR in this kind of typing package.

**Comment by alloy** [comment]
No worries!

You simply add your snippet of code that should work to this file https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/relay-runtime/relay-runtime-tests.tsx. As long as no type errors show up, it’s considered passing.

**Comment by renanmav** [comment]
@alloy I think thats it

**Comment by renanmav** [comment]
I could help split the remaining types.

**Comment by alloy** [comment]
Great, thanks!

As for splitting out further, that would be much appreciated 🙏
